File: 🟢 Easy/2273 - Find Resultant Array After Removing Anagrams/_2273FindResultantArrayAfterRemovingAnagrams.cpp
class Solution {
public:
    vector<string> removeAnagrams(vector<string>& words) {
        vector<string> res;
        string prev = "";

        for (int i = 0; i < words.size(); i++) {
            string curr = words[i];
            sort(curr.begin(), curr.end());

            if (curr != prev) {
                res.push_back(words[i]);
            }

            prev = curr;
        }

        return res;
    }
};

File: 🟢 Easy/2273 - Find Resultant Array After Removing Anagrams/_2273FindResultantArrayAfterRemovingAnagrams.java
class Solution {
    public List<String> removeAnagrams(String[] words) {
        List<String> res = new ArrayList<>();
        // 0 < i < words.length
        // words[i - 1] and words[i] are anagrams i.e. anargrams are present
        // consecutively.
        // delete words[i] from words (IMP)

        String prev = "";

        for (int i = 0; i < words.length; i++) {
            char charArr[] = words[i].toCharArray();
            Arrays.sort(charArr);
            // ARRAY --> STRING
            String curr = String.valueOf(charArr);

            if (!curr.equals(prev)) {
                res.add(words[i]);
            }

            prev = curr;
        }

        return res;
    }
}